source/faq.txt

```diff
@@ -148,6 +146,49 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+
+How do I prevent the "IllegalArgumentException: Invalid BSON field name" error?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your application may throw this exception if you pass an incorrectly formatted
+document to an operation and you are using a driver version v4.7 or earlier.
+In driver versions v4.8 and later, this error message was replaced by one
+that includes more specific details on what was incorrectly formatted.
+
+For example, the driver throws this error when you call an update operation
```

Comment:
With the first paragraph ending by talking about v4.8 and later, then the second paragraph starting with "For example...", it seems like the second paragraph (and the following code examples) are related to v4.8 and later, but if I'm understanding correctly they are talking about the error that occurs in v4.7 or earlier. 

I think there are a few ways we could mitigate this confusion. We could separate the sentence about v4.8 into an admonition and move it somewhere else (maybe at the end of the section). Or we could start the first paragraph with "In driver versions 4.8 and later...", that way the rest of the section is focused on the versions in which this error shows up. 

Hope that all makes sense, feel free to reach out if it's confusing.

Reply:
Thanks for identifying this. I moved it to a "note" admonition since it supplements info in certain cases.